charts/app-stable-diffusion/templates/deployment.yaml

```diff
@@ -36,62 +37,60 @@ spec:
             {{- end }}
           env:
-          {{- range $k, $v := .Values.deployment.env }}
+          {{- range $k, $v := .Values.api.env }}
             - name: {{ $k }}
               value: {{ $v | quote }}
           {{- end }}
-          command: [ "huggingface-cli" ]
+          command: [ "/bin/sh", "-c" ]
           args:
-            - download
-            - {{ .Values.model.modelHFName }}
-            {{- if .Values.model.modelFiles }}
-            - {{ .Values.model.modelFiles }}
-            {{- end }}
+{{/*            huggingface Docker was not updated for a long time so it could not recognize the dir param */}}
+            - "pip install -U 'huggingface_hub[cli]' && huggingface-cli download {{ .Values.model.modelHFName }} {{ if .Values.model.modelFiles }} {{ .Values.model.modelFiles }} {{ end }} --local-dir /home/worker/app/huggingface-cache/"
       {{- end }}
       securityContext:
         fsGroup: 0
       containers:
         - name: {{ .Chart.Name }}
-          image: "{{ .Values.deployment.image.repository }}:{{ .Values.deployment.image.tag }}"
-          imagePullPolicy: {{ .Values.deployment.image.pullPolicy }}
+          image: "{{ .Values.api.image.repository }}:{{ .Values.api.image.tag }}"
+          imagePullPolicy: {{ .Values.api.image.pullPolicy }}
           ports:
             - name: http
-              containerPort: {{ .Values.service.port }}
+              containerPort: {{ .Values.api.service.port }}
               protocol: TCP
-          {{- if .Values.deployment.healthChecks.enabled }}
           livenessProbe:
-            initialDelaySeconds: {{ .Values.deployment.healthChecks.delay }}
+            initialDelaySeconds: 60
             httpGet:
-              path: {{ .Values.deployment.healthChecks.endpoint }}
+              path: /sdapi/v1/options
               port: http
           readinessProbe:
-            initialDelaySeconds: {{ .Values.deployment.healthChecks.delay }}
+            initialDelaySeconds: 60
             httpGet:
-              path: {{ .Values.deployment.healthChecks.endpoint }}
+              path: /sdapi/v1/options
               port: http
-          {{- end }}
           resources:
-            {{- toYaml .Values.deployment.resources | nindent 12 }}
+            {{- toYaml .Values.api.resources | nindent 12 }}
           volumeMounts:
             - mountPath: /dev/shm
               name: dshm
             {{- if .Values.HFCacheVolume.enabled }}
             - name: huggingface-cache
-              mountPath: /home/worker/app/huggingface-cache
+              mountPath: {{ $huggingfaceDir }}
             {{- end }}
-          {{- if .Values.deployment.env }}
+          {{- if .Values.api.env }}
           env:
-          {{- range $k, $v := .Values.deployment.env }}
+          {{- range $k, $v := .Values.api.env }}
             - name: {{ $k }}
-              value: {{ $v | quote }}
+              value: |
+                {{- if eq $k "COMMANDLINE_ARGS" }}
```

Comment:
nit: why decided to go via env, not direct CLI args?

Reply:
removed the args